gpcontrib/gp_replica_check/gp_replica_check.py

```diff
@@ -58,7 +59,50 @@ def __str__(self):
 Mirror Data Directory Location: %s' % (self.getName(), self.host, self.port, self.datname,
                                           self.ploc, self.mloc)
 
+    def wait_for_wal_sync(self):
+        cmd = "PGOPTIONS='-c gp_role=utility' psql -h %s -p %s -d %s -t -A -c \"SELECT pg_current_wal_lsn() AS master_wal, replay_lsn AS standby_wal, pg_current_wal_lsn() = replay_lsn AS are_equal FROM pg_stat_replication;\"" % (self.host, self.port, pipes.quote(self.datname))
+        while True:
+            try:
+                output = subprocess.check_output(cmd, shell=True).decode().strip().split("\n")
+                print(f"Debug - Full output: {output}")  # Debug print
+
+                if not output:
```

Comment:
When no output received?

Reply:
Case1 : wal streaming is broken; 
Case2:  dest postgres is crashed.